### ISSUE TITLE
[Bug]: pdf parse ： memory leak ！

### ISSUE THREAD

**Issue by amzfc** [infiniflow/ragflow#4309]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### RAGFlow workspace code commit ID

26d3480e

### RAGFlow image version

26d3480e

### Other environment information

```Markdown
ubuntu 22.04
```

### Actual behavior

the matter of memory leak occurs when  pdf parseing.

### Steps to reproduce

```Markdown
using memory_profiler may find the bug ....
at the /deepdoc/parser/pdf_parser.py , the line 960..
![image](https://github.com/user-attachments/assets/ef09b370-2b78-4f18-a12e-644330f9b2b5)
finding the page_imagesv2 takes up a lot of memory,but it will not be released.
```

**Comment by amzfc** [comment]
/deepdoc/parser/pdf_parser.py , the line 960..
![image](https://github.com/user-attachments/assets/abbe83c0-b230-4d67-94a8-a8b5003bc8b5)

**Comment by KevinHuSh** [comment]
How many pages are there?

**Comment by amzfc** [comment]
@KevinHuSh 
This is about a 20-plus page PDF. And I need multiple PDFS to process.

**Comment by ODAncona** [comment]
@amzfc I bet this is related to #4031

It may be a duplicate

**Comment by amzfc** [comment]
@ODAncona 
Yes, the problem arises that the image occupied content during the PDF parsing process is not released

**Comment by stan-wang-analycia** [comment]
Any updates on this? thanks!
The issue persists even when using external embedding model. So I guess it's within the OCR or other steps?

**Comment by Yannnnnnny** [comment]
Since the issue has been resolved, I'll close this issue for now. Feel free to open a new issue if you run into any further problems.
